homeassistant/components/orangepi_gpio/switch.py

```diff
@@ -0,0 +1,86 @@
+"""Allows to configure a switch using Orange Pi GPIO."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.switch import PLATFORM_SCHEMA
+from homeassistant.components import orangepi_gpio
+from homeassistant.const import DEVICE_DEFAULT_NAME
+from homeassistant.helpers.entity import ToggleEntity
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['orangepi_gpio']
+
+CONF_PORTS = 'ports'
+CONF_PINMODE = 'pinmode'
+CONF_INVERT_LOGIC = 'invert_logic'
+
+PINMODES = ['pc', 'zeroplus', 'zeroplus2', 'deo', 'neocore2']
+
+DEFAULT_INVERT_LOGIC = False
+
+_SWITCHES_SCHEMA = vol.Schema({
+    cv.positive_int: cv.string,
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
```

Comment:
This Schema is identical to the one for the binary_sensor. Let's use one of them from a const file instead